### ISSUE TITLE
FactDollar still broken

### ISSUE THREAD

**Issue by tueda** [vermaseren/form#45]
In the following example, `$a = 1+x-y` is factorized as `$a[1] = -1` and `$a[2] = 1+x-y`. The overall sign is wrong.
```
S x,y;
#$a = 1+x-y;
#factdollar $a
#$b = 1
  #do i=1,`$a[0]'
    * (`$a[`i']')
  #enddo
;
#$c = $a - $b;  * should be 0
#message $a=`$a'
#do i=0,`$a[0]'
  #message $a[`i']=`$a[`i']'
#enddo
#message $b=`$b'
#message $c=`$c'
.end
```
```
~~~$a=1-y+x
~~~$a[0]=2
~~~$a[1]=-1
~~~$a[2]=x-y+1
~~~$b=-1+y-x
~~~$c=2-2*y+2*x
```

**Comment by vermaseren** [comment]
This was a missed special case.

Cheers

Jos

**Comment by tueda** [comment]
OK. I confirmed that it was fixed in 319e9ecbce359e06bd086ef014aae5d5d80675fa. Thanks.
